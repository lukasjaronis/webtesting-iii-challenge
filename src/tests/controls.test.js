import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Controls from '../controls/Controls';


test('Matches snapshot, fireevent on button close gate', () => {
    const { getByText, baseElement } = render(<Controls />);
    const button = getByText(/Close Gate/i)
    fireEvent.click(button);
    expect(baseElement).toMatchSnapshot(); 
})

test('Matches snapshot, fireevent on button lock gate', () => {
    const { getByText, baseElement } = render(<Controls />);
    const button = getByText(/Lock Gate/i)
    fireEvent.click(button);
    expect(baseElement).toMatchSnapshot(); 
})