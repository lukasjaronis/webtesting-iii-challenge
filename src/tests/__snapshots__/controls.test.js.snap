// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot, fireevent on button close gate 1`] = `
<body>
  <div>
    <div
      class="controls panel"
    >
      <button
        class="toggle-btn"
        disabled=""
      >
        Lock Gate
      </button>
      <button
        class="toggle-btn"
      >
        Close Gate
      </button>
    </div>
  </div>
</body>
`;

exports[`Matches snapshot, fireevent on button lock gate 1`] = `
<body>
  <div>
    <div
      class="controls panel"
    >
      <button
        class="toggle-btn"
        disabled=""
      >
        Lock Gate
      </button>
      <button
        class="toggle-btn"
      >
        Close Gate
      </button>
    </div>
  </div>
</body>
`;
